File: python/75_Sort_Colors.py
#!/usr/bin/python

class Solution(object):
    def sortColors(self, nums):
    	n = len(nums)
        i, left, right = 0, 0, n - 1
        while i < n:
        	tmp = nums[i]
        	if nums[i] == 0 and i >= left:
        		nums[left], nums[i] = tmp, nums[left]
        		left += 1
        		i += 1
        	elif nums[i] == 2 and i < right:
        		nums[right], nums[i] = tmp, nums[right]
        		right -= 1
        	else:
        		i += 1
def main():
	s = Solution()
	num = [2]
	s.sortColors(num)
	print num

if __name__ == "__main__":
    main()

File: c++/75_Sort_Colors.cpp
#include <iostream>
#include <vector>
#include <map>
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    void sortColors(vector<int>& nums) {
    	int n = nums.size();
    	int left = 0, right = n - 1, i = 0;
    	while (i <= right) {
    		if (nums[i] == 0)
    			swap(nums[i++], nums[left++]);
    		else if (nums[i] == 2)
    			swap(nums[i], nums[right--]);
    		else
    			i++;
    	}
    }
};


int main() {
	Solution s;
	//int dat[] = {0, 0, 1, 1, 0, 0, 2, 2, 0, 0, 1, 1, 2, 0};
	int dat[] = {2};
	vector<int> vec(dat, dat + sizeof(dat) / sizeof(int));
	s.sortColors(vec);
	for (int i = 0; i < vec.size(); i++)
		cout << vec[i] << " ";

}